src/server/routes/entity/entity.js

```diff
@@ -283,59 +411,52 @@ export function handleDelete(
 	const editorJSON = req.session.passport.user;
 	const {body}: {body: any} = req;
 
-	const entityDeletePromise = bookshelf.transaction((transacting) => {
+	const entityDeletePromise = bookshelf.transaction(async (transacting) => {
+		const otherEntities = await deleteRelationships(orm, transacting, entity);
+
 		const editorUpdatePromise =
 			utils.incrementEditorEditCountById(orm, editorJSON.id, transacting);
 
 		const newRevisionPromise = new Revision({
 			authorId: editorJSON.id
 		}).save(null, {transacting});
 
-		// Get the parents of the new revision
-		const revisionParentsPromise = newRevisionPromise
-			.then((revision) =>
-				revision.related('parents').fetch({require: false, transacting}));
-
-		// Add the previous revision as a parent of this revision.
-		const parentAddedPromise =
-			revisionParentsPromise.then(
-				(parents) => parents && parents.attach(
-					entity.revisionId, {transacting}
-				)
-			);
-
-		const notePromise = newRevisionPromise
-			.then((revision) => _createNote(
-				orm, body.note, editorJSON.id, revision, transacting
-			));
+		const newRevisionPromiseRevision = await newRevisionPromise;
 
 		/*
 		 * No trigger for deletions, so manually create the <Entity>Revision
 		 * and update the entity header
 		 */
-		const newEntityRevisionPromise = newRevisionPromise
-			.then((revision) => new RevisionModel({
-				bbid: entity.bbid,
-				dataId: null,
-				id: revision.get('id')
-			}).save(null, {
-				method: 'insert',
-				transacting
-			}));
 
-		const entityHeaderPromise = newEntityRevisionPromise
-			.then((entityRevision) => new HeaderModel({
-				bbid: entity.bbid,
-				masterRevisionId: entityRevision.get('id')
-			}).save(null, {transacting}));
+		const newEntityRevisionPromise = await new RevisionModel({
+			bbid: entity.bbid,
+			dataId: null,
+			id: newRevisionPromiseRevision.get('id')
+		}).save(null, {
+			method: 'insert',
+			transacting
+		});
+
+		const entityRevision = newEntityRevisionPromise;
+
+		const entityHeaderPromise = await new HeaderModel({
+			bbid: entity.bbid,
+			masterRevisionId: entityRevision.get('id')
+		}).save(null, {transacting});
+
+		const newRevision = await newRevisionPromise;
+
+		const mainEntity = await fetchOrCreateMainEntity(
+			orm, transacting, false, entity, entity.type
+		);
+
+		await saveEntitiesAndFinishRevision(orm, transacting, false, newRevision, mainEntity, otherEntities, editorJSON.id, body.note);
```

Comment:
Here you want something like `const savedMainEntity = await saveEntitiesAndFinishRevision(…`, because we want to return that savedMainEntity at the end.